pocs/camera/libasi.py

```diff
@@ -40,35 +39,47 @@ def __init__(self, library_path=None, *args, **kwargs):
                 locate the library.
             OSError: raises if the ctypes.CDLL loader cannot load the library.
         """
-        super().__init__(*args, **kwargs)
-        library = load_library(name='ASICamera2', path=library_path, logger=self.logger)
-        self._CDLL = library
-        self._version = self.get_SDK_version()
-
-    # Properties
-
-    @property
-    def version(self):
-        """ Version of the ZWO ASI SDK """
-        return self._version
+        super().__init__(name='ASICamera2', library_path=library_path, **kwargs)
 
     # Methods
 
-    def get_num_of_connected_cameras(self):
-        """ Get the count of connected ASI cameras """
-        count = self._CDLL.ASIGetNumOfConnectedCameras()  # Return type is int, needs no Pythonising
-        self.logger.debug("Found {} connected ASI cameras".format(count))
-        return count
-
     def get_SDK_version(self):
         """ Get the version of the ZWO ASI SDK """
         # First set return type for function to pointer to null terminated string
         self._CDLL.ASIGetSDKVersion.restype = ctypes.c_char_p
         version = self._CDLL.ASIGetSDKVersion().decode('ascii')  # Get bytes so decode to str
         version = version.replace(', ', '.')  # Format the version string properly
-        self.logger.debug("ZWO ASI SDK version: {}".format(version))
         return version
 
+    def get_cameras(self):
+        """Convenience function to get a dictionary of all currently connected camera string IDs
+        and their corresponding integer camera IDs.
+        """
+        n_cameras = self.get_num_of_connected_cameras()
+        if n_cameras == 0:
+            raise error.PanError("No ZWO ASI camera devices found")
+
+        # Get the IDs
+        cameras = {}
+        for camera_index in range(n_cameras):
+            # Can get IDs without opening cameras by parsing the name string
+            info = self.get_camera_property(camera_index)
+            model, _, string_id = info['name'].partition('(')
```

Comment:
Still think a comment showing what expected format is would be helpful, otherwise you have to do a little work to figure out why there is some parsing for the `(` and `)`.